code/modules/mob/living/simple_animal/hostile/mining/elites/elite.dm

```diff
@@ -322,23 +329,43 @@ While using this makes the system rely on OnFire, it still gives options for tim
 	var/turf/tumor_turf = get_turf(src)
 	if(loc == null)
 		return
-	for(var/tumor_range_turfs in RANGE_TURFS(12, tumor_turf))
-		if(get_dist(tumor_range_turfs, tumor_turf) == 12)
-			var/obj/effect/temp_visual/elite_tumor_wall/newwall
-			newwall = new /obj/effect/temp_visual/elite_tumor_wall(tumor_range_turfs, src)
-			newwall.activator = activator
-			newwall.ourelite = mychild
+	for(var/tumor_range_turfs in RANGE_EDGE_TURFS(ARENA_RADIUS, tumor_turf))
+		var/obj/effect/temp_visual/elite_tumor_wall/newwall
+		newwall = new /obj/effect/temp_visual/elite_tumor_wall(tumor_range_turfs, src)
+		newwall.activator = activator
+		newwall.ourelite = mychild
 
 /obj/structure/elite_tumor/proc/border_check()
-	if(activator != null && get_dist(src, activator) >= 12)
+	if(activator != null && get_dist(src, activator) >= ARENA_RADIUS)
 		activator.forceMove(loc)
 		visible_message("<span class='warning'>[activator] suddenly reappears above [src]!</span>")
 		playsound(loc,'sound/effects/phasein.ogg', 200, 0, 50, TRUE, TRUE)
-	if(mychild != null && get_dist(src, mychild) >= 12)
+	if(mychild != null && get_dist(src, mychild) >= ARENA_RADIUS)
 		mychild.forceMove(loc)
 		visible_message("<span class='warning'>[mychild] suddenly reappears above [src]!</span>")
 		playsound(loc,'sound/effects/phasein.ogg', 200, 0, 50, TRUE, TRUE)
 
+/obj/structure/elite_tumor/HasProximity(atom/movable/AM)
+	if(!isliving(AM))
```

Comment:
since you're already doing a ishuman and isrobot check, if the atom isn't a living type then those two istypes will catch it so this check can be removed

Reply:
Done